core/src/main/scala/cats/syntax/semigroup.scala

```diff
@@ -1,14 +1,16 @@
 package cats
 package syntax
 
+import cats.macros.Ops
+
 trait SemigroupSyntax {
   // TODO: use simulacrum instances eventually
   implicit def semigroupSyntax[A: Semigroup](a: A): SemigroupOps[A] =
     new SemigroupOps[A](a)
 }
 
-class SemigroupOps[A](lhs: A)(implicit A: Semigroup[A]) {
-  def |+|(rhs: A): A = A.combine(lhs, rhs)
-  def combine(rhs: A): A = A.combine(lhs, rhs)
-  def combineN(rhs: Int): A = A.combineN(lhs, rhs)
+class SemigroupOps[A: Semigroup](lhs: A) {
+  def |+|(rhs: A): A = macro Ops.binop[A, A]
```

Comment:
Sorry, I still don't have a good picture of what Cats is doing with Simulacrum and/or Machinist. What's the advantage of using `macro Ops.binop[A, A]` instead of the obvious `A.combine(lhs, rhs)`?